### PR TITLE
Add hard limit for log filesize

### PR DESCRIPTION
**Pull request by agalitsyn**
add 1.5mb limit for log file, instead read whole file

### CODE REVIEW

File: cmd/wstail/tail.go [hunk 1]
```diff
@@ -35,6 +35,14 @@ const tailMaxDepth = 100
 // rotated log file as named by savelog
 const rotatedLogUncompressed = "messages.0"
 
+const (
+	Byte     = 1.0
```
Comment (a-palchikov) [inline cmd/wstail/tail.go]: Do you need floats just to be able to compute 1.5Mb?

File: cmd/wstail/tail.go [hunk 2]
```diff
@@ -46,18 +54,28 @@ func tailer(ws *websocket.Conn, filter filter) {
...
-	if filter.isEmpty() {
-		// Limit the output of an empty filter to last tailMaxDepth lines
-		tailingDepth = []string{"-n", fmt.Sprintf("%v", tailMaxDepth)}
-	}
 	var files []string
 	if rotated.Main != "" {
 		files = append(files, rotated.Main)
 	}
-	files = append(files, "-F", filePath)
+	files = append(files, "--follow", filePath)
```
Comment (a-palchikov) [inline cmd/wstail/tail.go]: `-F` is not the same as `--follow`

File: cmd/wstail/tail.go [hunk 3]
```diff
@@ -46,18 +54,28 @@ func tailer(ws *websocket.Conn, filter filter) {
...
+
+	f, err := os.Stat(filePath)
+	if err != nil {
+		log.Errorf("cannot get file info for %v: %v", filePath, err)
+		return
```
Comment (a-palchikov) [inline cmd/wstail/tail.go:64]: Why exit if the file is not (yet) there?

File: cmd/wstail/tail.go [hunk 4]
```diff
@@ -46,18 +54,28 @@ func tailer(ws *websocket.Conn, filter filter) {
...
+	}
+	size := float32(f.Size())
+	tailingDepth := []string{"--lines", "+1"}
+	var isTrimmed bool
+	if filter.isEmpty() || size > tailFileSizeLimit {
```
Comment (a-palchikov) [inline cmd/wstail/tail.go]: Cannot really use initial log file size as a restriction - the size restriction should ideally be applied to the results after the filter has been applied.

File: cmd/wstail/tail.go (hunk 2 above)
Comment (UnderGreen) [inline cmd/wstail/tail.go]: @a-palchikov with `--follow` you don't need to use man or help for command to understand what it means.

Comment (agalitsyn) [reply cmd/wstail/tail.go]: true,
```
man tail

-F     same as --follow=name --retry
```

I always try to use long names for options in code for readability.

File: cmd/wstail/tail.go (hunk 3 above)
Comment (agalitsyn) [reply cmd/wstail/tail.go:64]: thx for the highlighting this, I I did not know that it should work such way

File: cmd/wstail/tail.go (hunk 1 above)
Comment (agalitsyn) [reply cmd/wstail/tail.go]: not really, set of consts is easy to read

File: cmd/wstail/tail.go (hunk 4 above)
Comment (agalitsyn) [reply cmd/wstail/tail.go]: yes, but how achieve this result? pipe goes to scanner, which read it line by line

**Comment by agalitsyn** [comment]
I've fixed review comments

**Review by sofuture — APPROVED** [review]
LGTM

**Comment by a-palchikov** [comment]
Querying file size is useless for enforcing this limit.
Besides, size limit should have been applied to the (history + grep) output - not the tail. Limiting tail works w/o regard to the filtered contents and results in outputs much shorter than the advertised limit (100 lines)

**Comment by agalitsyn** [comment]
Thx for clarifying how it exactly should work, I will change behaviour

**Comment by agalitsyn** [comment]
I will open another PR since it's already merged, this stuff will be reverted

**Pull request merged**